src/require-strict-assert.ts

```diff
@@ -0,0 +1,80 @@
+// require-strict-assert.ts
+
+/*
+ * Copyright (c) 2021-2023 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import type { Rule } from 'eslint';
+
+export default {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Require importing strict version of node:assert and using non-strict assert functions.',
+      url: 'https://github.com/checkdigit/eslint-plugin',
+    },
+    fixable: 'code',
+  },
+  create(context) {
+    return {
+      Literal(node) {
+        if (
+          (node.value === 'node:assert' || node.value === 'node:assert/strict') &&
+          node.parent.type === 'ImportDeclaration'
+        ) {
+          const importDeclaration = node.parent;
+          const defaultSpecifier = importDeclaration.specifiers.find(
+            (specifier) => specifier.type === 'ImportDefaultSpecifier' || specifier.type === 'ImportNamespaceSpecifier',
+          );
+
+          if (defaultSpecifier) {
+            const importDeclarationRange = importDeclaration.range ?? [0, 0];
+            let rangeToReplace = defaultSpecifier.range ?? importDeclarationRange;
+            let correctedText = `{ strict as ${defaultSpecifier.local.name} }`;
+
+            if (node.value === 'node:assert/strict' && defaultSpecifier.range && importDeclaration.source.range) {
+              rangeToReplace = [defaultSpecifier.range[0], importDeclaration.source.range[1]];
+              correctedText = `{ strict as ${defaultSpecifier.local.name} } from 'node:assert'`;
+            }
+
+            context.report({
+              node: importDeclaration,
+              message: 'Invalid form of strict assertion mode',
+              fix: (fixer) => fixer.replaceTextRange(rangeToReplace, correctedText),
+            });
+          }
+        }
+      },
+      CallExpression(node) {
+        const callee = node.callee;
+        if (
+          callee.type === 'MemberExpression' &&
+          callee.object.type === 'Identifier' &&
+          callee.object.name &&
```

Comment:
since we're not checking the object.name is coming from the node:assert module, this will pick up false positives I'm thinking